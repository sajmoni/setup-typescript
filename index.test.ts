import path from 'node:path'
import { expect, test } from 'vitest'
import { execa } from 'execa'
import { temporaryDirectory } from 'tempy'
import { loadJsonFile } from 'load-json-file'
import { PackageJson, readPackage } from 'read-pkg'
import { getBinPath } from 'get-bin-path'

import tsConfig from './src/tsConfigTemplate'

test('setup-typescript', async () => {
  const binPath = await getBinPath()
  if (!binPath) {
    throw new Error('Bin path not found')
  }
  const directory = temporaryDirectory({ prefix: 'hello-world' })

  const { stdout } = await execa(binPath, [], {
    cwd: directory,
    env: {
      // @ts-expect-error
      FORCE_COLOR: 2,
    },
  })

  const generatedTsConfig = await loadJsonFile(
    path.join(directory, 'tsconfig.json'),
  )

  expect(generatedTsConfig).toEqual(tsConfig)

  const packageJson = (await readPackage({
    cwd: directory,
    normalize: false,
  })) as PackageJson & {
    devDependencies: Record<string, string>
  }

  expect(Object.keys(packageJson.devDependencies)).toEqual([
    'tsconfig-one',
    'typescript',
  ])

  console.log('stdout', stdout)
  expect(stdout).toMatchSnapshot()
})
