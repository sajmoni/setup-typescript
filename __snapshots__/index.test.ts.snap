// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`add-typescript 1`] = `
"
[34m[1m add-typescript[22m[39m

 [32mtypescript and tsconfig.json added![39m"
`;

exports[`setup-typescript 1`] = `
"
[34m[1m setup-typescript[22m[39m

 [32mtypescript and tsconfig.json added![39m"
`;
